www/js/coolhole-util.js

```diff
@@ -41,6 +41,25 @@ function coolholePostFormatMessage(data, last, div, time, name, message) {
     }
 }
 
+/**
+ * Additional logic after the chat message is appended to the message bugger, such as returnfire, emote overlaps.
+ * @param {*} data chat message object
+ * @param {*} div jquery div object of the entire message
+ * @param {*} safeUsername username
+ */
+function coolholePostAddChatMessage(data, div) {
+    try {
+        // Jquery object; the span that contains the chat message only
+        let jqueryChatSpan = div.children("span").last();
+
+        checkReturnFire(div, jqueryChatSpan);
+        checkOverlapEmotes(jqueryChatSpan);
+    }
+    catch(ex) {
+        console.error(ex);
+    }
+}
```

Comment:
If you're only talking about the if statement for the "chat-shadow" class, I posted a video in discord #dev channel showing why that if statement is there.

If you're talking about the overall architecture for "coolholePostAddChatMessage", it could be refactored to be  server side logic, but i don't know if its worth it. The returnFire's can be put on serverside probably, but the overlapping emotes probably can't. 

I can try a seperate branch to see how the returnFire's would look on serverside if you want.

Reply:
Per discord conversation, moving logic to serverside could work, but streeeggs and I both agreed it wasn't worth it for the time being.
I was able to change the "chat-shadow" check to checking the shadowmute status on the userlist. So this way, we atleast checking a boolean status instead of a css class.